src/DataProtection/DataProtection/src/KeyManagement/DefaultKeyResolver.cs

```diff
@@ -41,33 +46,71 @@ internal sealed class DefaultKeyResolver : IDefaultKeyResolver
     /// </remarks>
     private readonly TimeSpan _maxServerToServerClockSkew;
 
-    public DefaultKeyResolver()
-        : this(NullLoggerFactory.Instance)
+    public DefaultKeyResolver(IOptions<KeyManagementOptions> keyManagementOptions)
+        : this(keyManagementOptions, NullLoggerFactory.Instance)
     { }
 
-    public DefaultKeyResolver(ILoggerFactory loggerFactory)
+    public DefaultKeyResolver(IOptions<KeyManagementOptions> keyManagementOptions, ILoggerFactory loggerFactory)
```

Comment:
Style nit: might want to take advantage of primary constructors here 😄

Reply:
How does that work with overloaded constructors?  I find I tend to mostly find primary constructors helpful for pure data types, whereas this type is more complex.

Reply:
It's possible to use overloaded constructors with primary constructors. If

`public DefaultKeyResolver(IOptions<KeyManagementOptions> keyManagementOptions, ILoggerFactory loggerFactory)`

is your primary constructor, then you should be able to use `this` in the other overload without issue.

BTW: this and the collection expressions is optional feedback but I find them helpful language features to take advantage of.

Reply:
Personally, I don't find that clarifying here.